sdk/python/feast/infra/offline_stores/file.py

```diff
@@ -224,3 +230,13 @@ def pull_latest_from_table_or_query(
         )
 
         return table
+
+    @staticmethod
+    def __prepare_path(path: str):
+        if path.startswith("s3://"):
+            s3 = fs.S3FileSystem(
+                endpoint_override=os.environ.get("FEAST_S3_ENDPOINT_URL")
```

Comment:
With the new Feast SDK we're purely using the `feature_store.yaml` for configuration. I'd want to see if we can avoid introducing env vars as backdoors for configuration. Do you want to take a stab at adding a configuration option for this override, or even exclude the override for now?

Reply:
@woop I really need ```endpoint_override``` thus I will pass it to ```feature_store.yaml``` as ```s3_endpoint_url``` parameter.

Reply:
On second thought It's actually not clear where this configuration should be. I think there is a strong case for the configuration to be in the file source itself. If you add it to `feature_store.yaml` then it would probably have to be at the "offline store" level, which would imply a global configuration for all file sources.

Reply:
@woop You are right I also thought about this. What do you think maybe we should expose ```filesystem``` argument as ```pyarrow.parquet.read_table``` do ? I think It would also give HDFS support (I have seen such request in Issues).

Reply:
@woop Finally I have removed the ```FEAST_S3_ENDPOINT_URL`` usage. I have noticed that pyarrow supports endpoint override using proper path eg. ```s3://access_key:secret_key@mybucket/data.parquet?scheme=http&endpoint_override=host:port``` (HDFS is also supported). I hope that such solution is ok ?

Reply:
I like the idea, but can you be more specific about where you want this argument and how it would be different from the URI? I think what we need is to have key/value options that can be filled for specific filesystem types, like custom S3 endpoints for example.

Reply:
The example usage for s3 could be:

* For local minio s3 instance:
```
driver_hourly_stats = FileSource(
    path="s3://data/driver_stats.parquet?scheme=http&endpoint_override=localhost:9000",
    event_timestamp_column="datetime",
    created_timestamp_column="created",
)
```

* For AWS S3:
```
driver_hourly_stats = FileSource(
    path="s3://data/driver_stats.parquet",
    event_timestamp_column="datetime",
    created_timestamp_column="created",
)
```

The path format is parsed internally by pyarrow.

Reply:
In that case, wouldn't it make more sense to have an `ArrowSource()`? I'd not want Arrow arguments to leak through to our API unnecessarily. Alternatively we could have top level configuration fields instead of having them inside the path field.

Reply:
I see your point of view @woop. What about ```S3FileSource``` and ```HdfsFileSource``` which will inherit from ```FileSource``` ? Probably we shouldn't mix all options specific for different protocols in top level configuration in ```FileSource```.

Reply:
That sounds reasonable to me.

Reply:
Finally I decided to add ```s3_endpoint_override``` to ```FileSource```. Comment for ```FileSource``` path already was pointing that ```s3://,gs://,file://``` is supported thus I think that optional argument for s3 will be better than adding additional classes like ```S3FileSource```. What do you think ?

Reply:
I think its fine for now.